File: backend/app/schemas/payment.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, Field

from app.models.payment import PaymentConcept, PaymentMethod


class PaymentCreate(BaseModel):
    amount: Decimal = Field(gt=0, decimal_places=2)
    payment_date: date = Field(default_factory=date.today)
    method: PaymentMethod
    concept: PaymentConcept
    reference: str | None = None


class PaymentRead(PaymentCreate):
    id: int
    reservation_id: int

    model_config = {"from_attributes": True}

File: frontend/src/types/payment.ts
export type PaymentMethod =
  | "cash"
  | "transfer";

export type PaymentConcept =
  | "deposit"
  | "final_payment"
  | "additional_charges"
  | "extra_hours"
  | "damages";

export interface PaymentCreate {
  amount: number;
  payment_date: string;
  method: PaymentMethod;
  concept: PaymentConcept;
  reference: string | null;
}

export interface Payment
  extends PaymentCreate {
  id: number;
  reservation_id: number;
  receipt_original_name: string | null;
  receipt_content_type: string | null;
  receipt_url: string | null;
}